pkg/services/ngalert/store/instance_database.go

```diff
@@ -197,3 +198,39 @@ func (st DBstore) DeleteAlertInstancesByRule(ctx context.Context, key models.Ale
 		return err
 	})
 }
+
+func (st DBstore) FullSync(ctx context.Context, instances []models.AlertInstance) error {
```

Comment:
Can we get a test for this in `instance_database_test.go`? The code looks reasonable, but it would be nice to have a test to be sure and also avoid future regressions.

Reply:
Good point, will add one.